archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/CodeUnitsShouldTest.java

```diff
@@ -277,6 +285,345 @@ public void throwable_type_predicates(ArchRule rule, Collection<String> expected
+        );
+    }
+
+    @Test
+    @UseDataProvider("restricted_method_calls_by_classes_rules")
+    public void restricted_method_calls_by_classes_predicate(ArchRule rule) {
+        EvaluationResult result = rule.evaluate(importClasses(ClassWithMethodAndConstructor.class, ClassCorrectlyCallingMethodAndConstructor.class,
+                ClassWronglyCallingMethodAndConstructor.class));
+
+        assertThat(singleLineFailureReportOf(result))
+                .contains("Rule 'methods should only be called by classes that")
+                .containsPattern(String.format("Method <%s.%s> calls method <%s.%s>",
+                        quote(ClassWronglyCallingMethodAndConstructor.class.getName()),
+                        quote(METHOD_CALL_WRONGLY),
+                        quote(ClassWithMethodAndConstructor.class.getName()),
+                        quote(METHOD_ONE_ARG)));
+    }
+
+    @DataProvider
+    public static Object[][] restricted_code_unit_calls_by_classes_rules() {
+        return $$(
+                $(codeUnits().should(onlyBeCalledByClassesThat(equivalentTo(ClassCorrectlyCallingMethodAndConstructor.class)))),
+                $(codeUnits().should().onlyBeCalled().byClassesThat(equivalentTo(ClassCorrectlyCallingMethodAndConstructor.class))),
+                $(codeUnits().should().onlyBeCalled().byClassesThat().belongToAnyOf(ClassCorrectlyCallingMethodAndConstructor.class))
+        );
+    }
+
+    @Test
+    @UseDataProvider("restricted_code_unit_calls_by_classes_rules")
+    public void restricted_code_units_calls_by_classes_predicate(ArchRule rule) {
+        EvaluationResult result = rule.evaluate(importClasses(ClassWithMethodAndConstructor.class, ClassCorrectlyCallingMethodAndConstructor.class,
+                ClassWronglyCallingMethodAndConstructor.class));
+
+        assertThat(singleLineFailureReportOf(result))
+                .contains("Rule 'code units should only be called by classes that")
+                .containsPattern(String.format("Method <%s.%s> calls method <%s.%s>",
+                        quote(ClassWronglyCallingMethodAndConstructor.class.getName()),
+                        quote(METHOD_CALL_WRONGLY),
+                        quote(ClassWithMethodAndConstructor.class.getName()),
+                        quote(METHOD_ONE_ARG)))
+                .containsPattern(String.format("Method <%s.%s> calls constructor <%s.%s>",
+                        quote(ClassWronglyCallingMethodAndConstructor.class.getName()),
+                        quote(METHOD_CALL_WRONGLY),
+                        quote(ClassWithMethodAndConstructor.class.getName()),
+                        quote(CONSTRUCTOR_ONE_ARG)));
+    }
+
+    @DataProvider
+    public static Object[][] restricted_code_unit_calls_by_methods_rules() {
+        return $$(
+                $(codeUnits().should().onlyBeCalled().byMethodsThat(declaredIn(ClassCorrectlyCallingMethodAndConstructor.class))),
```

Comment:
What about switching the order of the two data points so it matches the ones before where `onlyBeCalledByClassesThat` always comes first?